### PR TITLE
Grant admin group full access to all admin sets and all workflows

### LINKED ISSUE #1142: Automatically add 'admin' group to Admin Set Managers so repository admins can manage all Admin Sets
As an admin user, I should be able to update a work's state even if I haven't explicitly been granted a role in that workflow or in the work's Admin Set. Users who already have this ability include:

* The admin user that created the Admin Set
* Any user listed as a Manager for the Admin Set

It **excludes** any repository admin other than the one that created the Admin Set. To complete this ticket:

* Add the 'admin' group to the managers of every admin set when it's created.
* In the Admin Set edit view (Participants tab), display the 'admin' group in the list of Managers but prevent it from being removed from this list (both in the view and in the back-end).

**NOTE**: as a courtesy to users upgrading from earlier Hyrax versions, include in this work a new rake task that adds the 'admin' group to the list of Managers for all Admin Sets (except for those that already have it).

# Related work

See also: https://github.com/samvera-labs/hyrax/issues/1094
Decomposed from #185

### PR DESCRIPTION
**Pull request by mjgiarlo**
And prevent users from removing this group.

Fixes #1142 

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by jcoyne** [comment]
@ggeisler @hannahfrost is it possible that the admins might not want to be a depositing member of every admin set?  That could make this list really long:

screen shot 2017-06-09 at 3 20 29 pm

**Comment by jcoyne** [comment]
What if the implementation isn't using the group named `admin`?  Isn't this the case for Scholarsphere?

**Comment by mjgiarlo** [comment]
@awead @cam156 Thoughts on @jcoyne's question above?

**Comment by awead** [comment]
@mjgiarlo we have an `administrator?` method on `User`:
https://github.com/psu-stewardship/scholarsphere/blob/develop/app/models/user.rb#L79

**Comment by mjgiarlo** [comment]
@awead OK. The latest push to this branch includes the ability to set a custom admin group name, such as `'umg/up.dlt.scholarsphere-admin-viewers'`, so this should be open for y'all to make it work for da sphere.

**Comment by mjgiarlo** [comment]
@jcoyne Ready for re-review. Now includes the UI work and a new rake task.

File: spec/services/hyrax/admin_set_create_service_spec.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 RSpec.describe Hyrax::AdminSetCreateService do
+  before do
+    # Load expected Sipity roles, which were likely cleaned by DatabaseCleaner
+    Hyrax.config.persist_registered_roles!
```
Comment (jcoyne) [inline spec/services/hyrax/admin_set_create_service_spec.rb]: Can this move this down to line 65?

File: spec/controllers/hyrax/generic_works_controller_json_spec.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@
   routes { Rails.application.routes }
 
   let(:user) { create(:user) }
-  before { sign_in user }
+
+  before do
```
Comment (jcoyne) [inline spec/controllers/hyrax/generic_works_controller_json_spec.rb]: This change shouldn't be part of this PR.

File: spec/controllers/hyrax/admin/permission_template_accesses_controller_spec.rb [hunk 3]
```diff
@@ -23,20 +23,37 @@
...
-        expect(flash[:notice]).to eq(I18n.t('participants', scope: 'hyrax.admin.admin_sets.form.permission_update_notices'))
-        expect(admin_set.reload.edit_users).to be_empty
+      context 'when deleting the admin users group' do
+        let(:agent_type) { 'group' }
+        let(:agent_id) { 'admin' }
+        it "fails" do
+          expect(controller).to receive(:authorize!).with(:destroy, permission_template_access)
+          expect do
+            delete :destroy, params: { id: permission_template_access }
+          end.to change { Hyrax::PermissionTemplateAccess.count }.by(0)
```
Comment (jcoyne) [inline spec/controllers/hyrax/admin/permission_template_accesses_controller_spec.rb]: `.not_to change { ... }` ?

File: spec/controllers/hyrax/generic_works_controller_json_spec.rb (hunk 2 above)
Comment (mjgiarlo) [inline spec/controllers/hyrax/generic_works_controller_json_spec.rb]: Oops. This change was from back when I had the `persist_regisered_roles` changes in the branch. Will remove.

**Comment by mjgiarlo** [comment]
@jcoyne All requested changes made.
